module/apmotelbridge/tracer.go

```diff
@@ -0,0 +1,157 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package apmotelbridge // import "go.elastic.co/apm/module/apmotelbridge/v2"
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"go.elastic.co/apm/v2"
+	"go.opencensus.io/trace"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+// Tracer creates a named tracer that implements otel.Tracer.
+// If the name is an empty string then provider uses default name.
+func Tracer(name string, opts ...trace.TracerOption) *tracer {
+	if name == "" {
+		name = "otel_bridge"
+	}
+
+	apmOpts := apm.TracerOptions{ServiceName: name}
+	if version := trace.NewTracerConfig(opts).InstrumentationVersion(); version != "" {
+		apmOpts.ServiceVersion = version
+	}
+
+	t := apm.NewTracerOptions(apmOpts)
```

Comment:
Can we make it so an `apm.Tracer` can be passed in, enabling all the customisation one may desire. Perhaps `NewTracerProvider(*apm.Tracer)`?

OTOH, I think it should also be possible to pass in a `*resource.Resource` (ala sdktrace.WithResource), which would be used to set the service name, version, environment, global labels, etc. In our API, "resource" attributes are associated with `*apm.Tracer`. Not too sure off hand how to reconcile those two things.

Also: I could be mistaken, but I don't think the tracer name is expected to correspond to the service name.

Reply:
>Also: I could be mistaken, but I don't think the tracer name is expected to correspond to the service name.

Roger. It doesn't seem then like we have a use for the OTel tracer name? Since apm doesn't name the tracer, only the service.

Reply:
Yeah I don't think we have a use for it at the moment.